include/opendht/indexation/pht.h

```diff
@@ -166,6 +175,125 @@ class Pht {
     void insert(Key k, Value v, Dht::DoneCallbackSimple cb = {});
 
 private:
+    class Cache {
+    public:
+        /**
+         * Insert all needed node into the tree according to a prefix
+         * %param p : Prefix that we need to insert
+         */
+        void insert(const Prefix& p) {
+            size_t i = 0;
+            auto now = clock::now();
+
+            std::shared_ptr<Node> curr_node;
+
+            while ( ( leaves_.size() > 0 && leaves_.begin()->first + NODE_EXPIRE_TIME < now )
+                    || leaves_.size() > MAX_ELEMENT ) {
+
+                leaves_.erase(leaves_.begin());
+            }
+
+            if ( !(curr_node = root_.lock()) ) {
+
+                /* Root does not exist, need to create one*/
+                curr_node = std::make_shared<Node>();
+                root_ = curr_node;
+            }
+            
+            curr_node->last_reply = now;
+
+            /* Iterate through all bit of the Blob */
+            for ( i = 0; i < p.size_; i++ ) {
+
+                /* According to the bit define which node is the next one */
+                auto& next = ( p.getBit(i) ) ? (*curr_node).right_child : (*curr_node).left_child;
```

Comment:
You should use the syntax `pointer->element` instead of `(*pointer).element`.
It's simpler and more consistent with the rest of opendht code.